pkgs/os-specific/linux/kernel/common-config.nix

```diff
@@ -213,6 +213,10 @@ let
       MPTCP           = whenAtLeast "5.6" yes;
       MPTCP_IPV6      = whenAtLeast "5.6" yes;
       INET_MPTCP_DIAG = whenAtLeast "5.9" (mkDefault module);
+
+      # Kernel TLS
+      TLS         = whenAtLeast "4.13" yes;
```

Comment:
Does this built as module or into the kerne?

Reply:
Supports both variants. Thought it would be better to build directly to kernel.

Reply:
If there's a modular option, we should always use that unless there's a good reason not to.  Otherwise it consumes memory and disk space (and impacts boot time because of that) for anybody who's not using it.

Reply:
Fixed.

Reply:
If use a hardened profile, the TLS module is not automatically loaded.
Kernel size when using a TLS module: - 6562752 bytes (6,3M), without module - 6612512 bytes (6,4M). The size differs by only 48 kb.

Reply:
@ajs124 @alyssais can I try to change build without a module? :)

Reply:
> If use a hardened profile, the TLS module is not automatically loaded.

What do you mean by a hardened profile in this context?  A hardened kernel?

Reply:
Yes, hardened kernel + profile `<nixpkgs/nixos/modules/profiles/hardened.nix>`
```
  imports = [
    <nixpkgs/nixos/modules/profiles/hardened.nix>
  ];
```

Reply:
iirc that only disables module loading after bootup, so you can still just put this module into `boot.kernelModules`

Reply:
I do it myself.
I would like the module to be automatically activated for other users, if there is support from the software.

Reply:
If users are deliberately opting out of automatic module loading, that's on them.  They can either load the module themselves, or they can turn on automatic loading.  It's not for us to guess what modules they might like to have loaded.